shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java

```diff
@@ -0,0 +1,172 @@
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+
+@ApplicationScoped
+@ControllerConfiguration(labelSelector = LabelsBuilder.V2_RECONCILER_LABEL_SELECTOR)
+public class ManagedBridgeController implements Reconciler<ManagedBridge>,
+        EventSourceInitializer<ManagedBridge>,
+        ErrorStatusHandler<ManagedBridge> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedBridgeController.class);
+
+    @ConfigProperty(name = "event-bridge.managed-bridge.poll-interval.milliseconds")
+    int reconcileIntervalSeconds;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    NetworkingService networkingService;
+
+    @Inject
+    ManagedBridgeService managedBridgeService;
+
+    @Inject
+    OperatorMetricsService metricsService;
+
+    @Override
+    public UpdateControl<ManagedBridge> reconcile(ManagedBridge managedBridge, Context context) {
+
+        LOGGER.info("Reconciling ManagedBridge: '{}' in namespace '{}'",
+                managedBridge.getMetadata().getName(),
+                managedBridge.getMetadata().getNamespace());
+
+        ManagedBridgeStatus status = managedBridge.getStatus();
+
+        //
+        //        if (!status.isReady() && isTimedOut(status)) {
+        //            notifyManagerOfFailure(bridgeIngress,
+        //                                   new ProvisioningTimeOutException(String.format(ProvisioningTimeOutException.TIMEOUT_FAILURE_MESSAGE,
+        //                                                                                  bridgeIngress.getClass().getSimpleName(),
+        //                                                                                  bridgeIngress.getSpec().getId())));
+        //            status.markConditionFalse(ConditionTypeConstants.READY);
+        //            return UpdateControl.updateStatus(bridgeIngress);
+        //        }
+
+        Secret secret = managedBridgeService.fetchOrCreateBridgeSecret(managedBridge);
+
+        if (secret == null) {
+            LOGGER.info("The Secret for the ManagedBridge with id '{}' has been not created yet.",
+                    managedBridge.getSpec().getId());
+            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
+                status.markConditionFalse(ConditionTypeConstants.READY);
+            }
+            if (!status.isConditionTypeFalse(ManagedBridgeStatus.SECRET_AVAILABLE)) {
+                status.markConditionFalse(ManagedBridgeStatus.SECRET_AVAILABLE);
+            }
+            return UpdateControl.updateStatus(managedBridge).rescheduleAfter(reconcileIntervalSeconds);
+        } else if (!status.isConditionTypeTrue(ManagedBridgeStatus.SECRET_AVAILABLE)) {
+            LOGGER.info("Secret for ManagedBridge with id '{}' has been created.", managedBridge.getSpec().getId());
+            status.markConditionTrue(ManagedBridgeStatus.SECRET_AVAILABLE);
```

Comment:
As discussed in MGDOBR-1186 design, if any component fails then we should break the flow and directly update status/send condition to manager.

If Secret creation is failed then it make no scence to create configmap and KnativeBroker.

Reply:
Yep. Please see comments about this being the "get it working" PR 👍